src/components/SearchFacet/SearchFacet.tsx

```diff
@@ -215,42 +252,163 @@ export const SearchFacets = (props: ISearchFacetsProps) => {
+    setDraggingFacetId(event.active.id as SearchFacetID);
+  };
+
+  // Make the changes permanent
   const handleDragEnd = (event: DragEndEvent) => {
     const { active, over } = event;
 
     if (!over) {
       return;
     }
 
-    if (active?.id !== over?.id) {
-      const oldIndex = facets.indexOf(active?.id as SearchFacetID);
-      const newIndex = facets.indexOf(over?.id as SearchFacetID);
-      setFacets(arrayMove(facets, oldIndex, newIndex));
+    const { visible } = facetsList;
+
+    if (visible.length < facets.length) {
+      // item moved to hidden
+      hideSearchFacet(active.id as SearchFacetID);
+    } else if (visible.length > facets.length) {
+      // item moved to visible
+      showSearchFacet(
+        active.id as SearchFacetID,
+        visible.findIndex((id) => id === active.id),
+      );
+    } else {
+      // item order moved
+      setFacets([...visible]);
     }
   };
 
-  const list = useMemo(() => {
-    return facets.map((facetId) => {
+  const visibleItems = useMemo(() => {
+    return facetsList.visible.map((facetId) => {
       const facetProps = facetConfig[facetId];
-      return <SearchFacet {...facetProps} key={facetProps.storeId} onQueryUpdate={onQueryUpdate} />;
+      return <SearchFacet {...facetProps} key={facetProps.storeId} onQueryUpdate={onQueryUpdate} hidden={false} />;
+    });
+  }, [facetsList.visible, onQueryUpdate]);
+
+  const hiddenItems = useMemo(() => {
+    const facetProps = facetsList.hidden.map((id) => facetConfig[id]).sort((a, b) => a.label.localeCompare(b.label));
+    return facetProps.map((facetProp) => {
+      return <SearchFacet {...facetProp} key={facetProp.storeId} onQueryUpdate={onQueryUpdate} hidden={true} />;
     });
-  }, [facets, onQueryUpdate]);
+  }, [facetsList.hidden, onQueryUpdate]);
+
+  const activeItem = useMemo(() => {
+    if (draggingFacetId) {
+      const facetProp = facetConfig[draggingFacetId];
+      return <SearchFacet {...facetProp} key={facetProp.storeId} onQueryUpdate={onQueryUpdate} hidden={true} />; // change hidden
+    }
+  }, [draggingFacetId]);
 
   return (
     <>
       <DndContext
-        modifiers={[restrictToVerticalAxis, restrictToParentElement]}
         sensors={[mouseSensor]}
+        onDragStart={handleDragStart}
+        onDragOver={handleDragOver}
         onDragEnd={handleDragEnd}
       >
-        <SortableContext items={facets} strategy={verticalListSortingStrategy}>
-          <List>{list}</List>
-        </SortableContext>
-      </DndContext>
-      {hidden.length > 0 && (
-        <Button variant="link" onClick={handleShowAllClick} type="button">
-          Show all ({hidden.length} hidden)
+        <DroppableContainer id="visible-container" itemsID={facetsList.visible} items={visibleItems} />
+
+        <Button variant="link" onClick={toggleShowHidden} type="button">
+          {showHiddenFacets ? 'Hide hidden filters' : 'Show hidden filters'} {`(${hiddenItems.length})`}
```

Comment:
I wonder if here it may be better to also have a chevron type toggle so that it's a clear indicator of the menu being open. 

When it's empty, you don't immediately know if it's open or not unless you actually read the text

Reply:
Added